### DISCUSSION TITLE
Do I need to set something specific in the User-Agent

### DISCUSSION THREAD

**Discussion by yiddytlq** [kyujin-cho/pixel-volte-patch#21] [Q&A]
VoLTE shows enabled by me but IMS is still unregistered. I am using the Pixel 7 Pro on Partner Communications which have VoLTE but just havent enabled it for my device.

They have a list of supported devices [here](https://www.partner.co.il/u/volte)
<details><summary>ENGLISH LIST HERE</summary>
<p>

Below is the updated list of models that support the VoLTE service:
 

SAMSUNG

Galaxy XCover6 Pro sub-model SM-G736B/DS starting from version: XXS1AVH5
Galaxy A04e sub-model SM-A042F/DS starting from version: XXU1AVJ2
Galaxy Z Flip4 5G sub-model SM-F721B starting from version: XXU1AVGA
Galaxy Z Fold 4 5G sub-model SM-F936B /DS starting from version: XXU1AVGA
13 Galaxy A submodel SM-A137F-DS starting from version: XXU1AVH1
Galaxy S21 FE submodel SM-G990E-DS starting from version: XXU2CVDA
Galaxy Note10 Lite submodel SM-N770F starting from version: XXU8GVD5
Galaxy Note10 Lite - Submodel SM-N770F starting with software version: XXU1ATA1
Galaxy A73 5G - submodel SM-A736B-DS starting with software version: XXU1AVC5
Galaxy A03 - submodel SM-A035F-DS starting with software version: XXU1AUL6
Galaxy A33 5G - submodel SM-A336E - DSN starting from software version: DXU1AVC9
Galaxy A23 - sub model SM-A235F-DS starting from software version: XXU1AVBA
Galaxy A13 - submodel SM-A135F-DS starting with software version: XXU1AVB5
Galaxy A53 5G - submodel SM-A536E-DS starting with software version: XXU1AVBB
Galaxy S22 Ultra - submodel SM-S908E starting with software version: XXU1AVB3
Galaxy S22 - submodel SM-S901E starting with software version: XXU1AVB3
Galaxy S22 Plus - submodel SM-S906E starting with software version: XXU1AVB3
Galaxy S21 FE 5G - submodel SM-G990E/DS starting with software version: XXU1BUKA
Galaxy M52 5G - submodel SM-M526BR-DS Starting with software version: XXU1AUI5
Galaxy A03s - submodel SM-A037F/DS Starting with software version: XXU1AUH3
Galaxy A52s 5G - submodel SM-A528B starting with software version: XXU1AUH3
Galaxy A12s - submodel SM-A127F/DS starting with software version: XXU2AUG9
Z Flip3 5G - sub model SM-F711B starting from software version: XXU1AUG7
Galaxy Z Fold3 5G - sub-model SM-F926B starting with software version: XXU1AUG8
Galaxy A52 5G - sub-model SM-A526B starting with software version: XXU1AUCA
Galaxy S21Ultra 5G - sub-model- SM-G998B starting with software version: XXU1ATLF
Galaxy S21 5G - sub-model - SM-G991B starting from software version: XXU1ATLF
Galaxy S21Plus 5G submodel - SM-G996B starting from software version: XXU1ATLD
Galaxy A02s submodel - SM-A025F/DS starting from software version - XXU1ATK4
Samsung Galaxy A32 submodel - SM-A325F starting from software version - XXU1AUC2
Galaxy A12 sub-model - SM-A125F/DS starting from software version - XXU1ATKC
Galaxy A52 sub-model - SM-A525F starting from software version - XXU1AUC4
Galaxy A72 sub-model - SM-A725F starting from software version - XXU1AUC4
Galaxy Note20 Ultra 5G sub-model - SM-N986B/DS starting from software version- XXU1ATJ1
Galaxy Xcover Pro sub-model - SM-G715FN starting from software version - XXU6ATK9
Samsung Galaxy S20 FE 5G - sub-model SM-G718B/DS starting from software version: XXU1ATIG
Samsung Galaxy Note20 Ultra - sub-model SM-N985F/DS starting from software version: XXU1ATGI
Samsung Galaxy Note20 - submodel SM-N980F/DS starting with software version: XXU1ATGK
Samsung Galaxy A11 - submodel SM-A115F/DS starting with software version: XXU1ATC5
Samsung Galaxy A31 - submodel SM-A315F/DS starting with software version: XXU1ATD4
Samsung Galaxy S20 Ultra LTE - sub model SM-G988B/DS starting from software version: XXU1ATB3
Samsung Galaxy S20 Plus - submodel SM-G985F/DS starting with software version: XXU1ATB3
Samsung Galaxy S20 - submodel SM-G980F/DS starting with software version: XXU1ATB3
Samsung Galaxy A71 - submodel SM-A715F starting with software version: XXU1ATA1
Samsung Galaxy A51 - Submodel SM-A515F starting with software version: XXU1ASL6 Samsung Galaxy A10s - Submodel
SM-A107F starting with software version: XXU3ASH6
Samsung Galaxy A30s - Submodel SM-A307FN starting with software version: XXU1ASHL 
Galaxy Note 10 Plus - submodel SM-N975F from version: XXU1ASGO
Galaxy Note10 - submodel SM-N970F from version: XXU1ASGO
Galaxy Samsung A80 - submodel SM-A805F from version: XXU1ASED
Samsung Galaxy A30 - submodel SM-A305F from version : DDU1ASBC
Samsung Galaxy A50 - sub-model SM-A505F starting from version: DDU1ASBE
Samsung Galaxy A70 - sub-model SM-A705FN starting from version: XXU1ASD1
Samsung Galaxy S10 Plus - sub-model SM-G975F starting from version: XXU1ASAT
Samsung Galaxy S10 - sub-model SM- G973F starting from version: XXU1ASAT
Samsung Galaxy S10e - submodel SM-G970F starting from version: XXU1ASAT
Samsung Galaxy A9 2018 - submodel SM-A920F starting from version XXU1ARKC
Samsung Galaxy A7 2018 - submodel SM-A750F from version XXU1ARII
Samsung Galaxy Note 9 - submodel SM-N960F from version XXU1ARG7
Samsung Galaxy J5 Pro - submodel SM-J530F from version XWU3BRJ1
Samsung Galaxy A5 2017 - submodel SM-A520F from Version XXU5CRGD
Samsung Galaxy J8 - submodel SM-J810F starting from version JXU1ARF2
Samsung Galaxy A8 2018 - submodel SM-A530F starting from version XXU2BRG3
Samsung Galaxy A8 Plus 2018 - submodel SM-A730F starting from version XXU2BRG5
Samsung Galaxy A7 2017 - submodel SM- A720F starting with version XXU3CRF3

Samsung Galaxy S7 - sub-model SM-G930F starting with version XXU2EREM

Samsung Galaxy S7 Edge - sub-model SM-G935F starting from version XXU2EREM

Samsung Galaxy S8 - SM-G950F sub-model starting from version XXU1CRB7

Samsung Galaxy S8 Plus - sub-model SM-G955F starting with version XXU1CRB7

Samsung Galaxy Note 8 - sub-model SM-N950F starting with version XXU3CRC1

Samsung Galaxy S9 - sub-model SM-G960F starting from version XXU1BRE5

Samsung Galaxy S9 Plus - sub-model SM-G965F starting with version XXU1BRE5

Samsung Galaxy J4 - SM-J400F sub-model starting from version XXU1ARE8

Samsung Galaxy J6 - SM-J600F sub-model starting from version XXU1ARE4

Nothing Phone

Nothing Phone (1) sub-model A063 starting from version: Spacewar-s1.1-221022-1221

BlackView

BlackView Tab 6 Kids starting from version: V1.0_2022514V05

 

PhoneLine

PhoneLine F33 starting from version: SHB_T117

APPLE

iPhone 14 sub-model A2882 from version: IOS 16.0-20A357 Operator version Partner 51.0
iPhone 14 Plus sub-model A2886 from version: IOS 16.0-20A357 Operator version Partner 51.0
iPhone 14 Pro sub-model A2890 from version: IOS 16.0-20A357 Operator version 51.0
iPhone 14 Pro Max sub-model A2894 starting from version: IOS 16.0-20A357 Operator version Partner 51.0
iPhone SE (2022) - sub-model A2783 starting from version 15.3.1
iPhone 13 pro max - sub-model A2643 starting from software version: iOS 15(19A346) Operator version 47.0
iPhone 13 pro - sub-model A2638 starting from software version: iOS 15(19A346) Operator version 47.0
iPhone 13 - sub-model A2633 starting from software version: iOS 15(19A346) Operator version 47.0
iPhone 13 mini - sub-model A2628 starting from software version : iOS 15(19A346) Operator version 47.0
iPhone 12 mini - submodel A2399 starting from software version: 14.2.1 operator version partner44.0
iPhone 12 - submodel A2403 starting from software version: 14.2.1 operator version partner44.0
iPhone 12 Pro - submodel A2407 starting from software version: 14.2. 1 operator version partner44.0
iPhone 12 Pro Max - submodel A2411 starting from software version: 14.2.1 operator version partner44.0
iPhone SE - submodel A2296 starting from software version: ios13.4 + operator version 40.0
iPhone 11 - starting from software version: 13.0 + operator version 37.0
iPhone 11 Pro Max - starting from software version: 13.0 + operator version 37.0
iPhone 11 Pro - starting from software version: 13.0 + operator version 37.0
iPhone 8 devices and above - starting from software version 11.3 + operator version 32.1
iPhone 6/6 devices Plus/6S/6S Plus/7/7 Plus - starting with software version 12.0 + operator version 33.0

LG

LG WING Sub-model: LM-F100EMW starting from software version - V10a
LG Velvet Sub-model: LM-G910EMW
LG Q7 PLUS Sub-model: LM-Q610AZ starting from version - V10d
LG G8S THINQ Sub-model: LM-G810EA starting from version: V10a
LG G7 ThinQ sub-model LM-G710EM starting from version: V10a

eTera

eTera E980 starting from version: V008_20191202

OnePlus

OnePlus Nord CE 5G - submodel EB2103 starting from software version: EB13AA.11.0.3.3 Build
OnePlus 8T - submodel KB2003 starting from software version: KB05AA.11.0.9.9 Build

OPPO

Build 11_A.11: starting from software version OPPO Reno7 5G - submodel CPH2371
Build 11_A.10: starting from software version CPH2343 submodel - OPPO Reno7 Z 5
Build 11_A.03: starting from software version OPPO A76 - submodel CPH2375
Build 11_A.12: Starting from software version OPPO Reno6 5G - submodel CPH225
Build 11_A.07: starting from software version OPPO A94 5G - submodel CPH2211

*Samsung S7/S8/Note 8 devices purchased at Cellcom or Pelephone will not work with the VoLTE service and it is required to burn a version at Partner Laboratories

</p>
</details>

Is there anything specific that I need to put in the User Agent string field?

**Comment by kyujin-cho** [comment]
Hi! Setting User-Agent for SIP varies heavily by carrier configuration, so there's no silver bullet for this. Moreover, in most of the case User-Agent only makes problem when trying to actually establish call, not just registering device to carrier's IMS server.

**Reply by victor-marino** [reply]
This is the current problem I have with VoWiFi. All enabled, but call drops when trying to establish it.

Do you know if there's a list of user-agents for popular devices? My carrier supports VoWiFi on a number of Samsung devices, so I could give that a shot.

**Reply by yiddytlq** [reply]
I hear you, is there anything else that I might have missed? IMS status is still unregistered.

**Reply by bige93** [reply]
> This is the current problem I have with VoWiFi. All enabled, but call drops when trying to establish it.
> 
> Do you know if there's a list of user-agents for popular devices? My carrier supports VoWiFi on a number of Samsung devices, so I could give that a shot.

Hi there! Did you have any success with that? Thanks!

**Reply by victor-marino** [reply]
Not yet, unfortunately.

I'm waiting for a friend who is on the same network and has a Samsung device to pull out the user-agent for me.

But from the tests I've done I don't think it's related to the user-agent. If you look at my comment [here](https://github.com/kyujin-cho/pixel-volte-patch/issues/24#issuecomment-1424560616), it looks like my phone is simply not able to register to my carriers IMS over Wi-Fi, because it cannot establish the tunnel towards the ePDG.

I'm still investigating but I suspect Google might need to adapt something to make it work with my carrier's ePDG.

**Comment by flightmansam** [comment]
Hi All, 

We have a massive problem with this in Australia as nationwide telcos are switching off the 3G network and forcing everyone to use VoLTE for calling. The telco providers (one in particular) are being absolute dicks about white listing only "modern" (i.e. their own) phones and are most likely* using the user agent string to either allow or deny third party devices on their networks. More info on this delimma attached here:
- https://www.aph.gov.au/Parliamentary_Business/Committees/Senate/Rural_and_Regional_Affairs_and_Transport/3GNetworkShutdown
- https://youtu.be/RPlTz-3estM?si=4k7pUvDGLERwoRpo
- https://www.reddit.com/r/australia/comments/1dhv8zv/do_you_have_a_xiaomi_phone_affected_by_the_3g/
- https://www.ozbargain.com.au/node/831732

So the DIY solution to this problem is to use the shizuku+pixel-volte-patch and also a custom user agent. For those who are new to android adb here is my results that might help with you. Most* android devices use the standard default template. 

### How do I get it?
Check to see your phone doesn't have something custom (prior to installing the pixel-volte-patch):
```bash
adb shell dumpsys carrier_config | grep agent
```
Most likely it will return: `ims.ims_user_agent_string = #MANUFACTURER#_#MODEL#_Android#AV#_#BUILD#`

### How do I complie the actual user agent string (based off a phone I know is working on the network of choice)?
As in the android dev page [here](https://developer.android.com/reference/android/telephony/CarrierConfigManager.Ims#KEY_IMS_USER_AGENT_STRING) you can complie the list through querying the android properties via adb:
MANUFACTURER :  `adb shell getprop ro.product.manufacturer`
MODEL :  `adb shell getpropro.product.model`
AV :  `adb shell getprop ro.build.version.release`
BUILD : `adb shell getprop ro.build.id`

You can then build the string as in the template. 

Please comment if this does/doesn't fix the problem. I will update this reply as I experiment with my success. 


\* will confirm

**Reply by vennand** [reply]
Hey @flightmansam ,

How do you actually modify the UserAgent field? Do you need to use adb? (I'd rather not, last time I tried, I failed to flash, and wrecked everything haha....)

I've just been told by my provider that they'll block my phone in 3 days, so spoofing the UserAgent might avoid this?

**Reply by flightmansam** [reply]
The pixel IMS app should have a field to change. If you want to try the user agent I'm using you could try this one: samsung_SM-F731B_Android14_UP1A.231005.007

I'm using this on a Sony Xperia 1 VI and haven't been getting the annoying Telstra messages. Please let me know if this works!!!

**Reply by vennand** [reply]
I'll give that a shot, thanks. I still get the message, but I'm assuming it's because they haven't updated their records. I use a OnePlus 8.

I'll let you know if it works. They said they would block it on November 28th, 2024. If it's not, then I guess it works haha

**Reply by vennand** [reply]
FYI they didn't block it, but I'm not sure it's because the signal was spoofed. If I send the SMS "3g" to the number 3498 (the 3G checker), it still reports that my phone is a OnePlus 8.
